rustls/src/client/hs.rs

```diff
@@ -1197,10 +1197,6 @@ fn process_cert_type_extension(
             AlertDescription::HandshakeFailure,
             Error::PeerIncompatible(PeerIncompatible::IncorrectCertificateTypeExtension),
         )),
-        (_, Some(CertificateType::RawPublicKey)) => Err(common.send_fatal_alert(
-            AlertDescription::HandshakeFailure,
-            Error::PeerIncompatible(PeerIncompatible::UnsolicitedCertificateTypeExtension),
-        )),
```

Comment:
Removing these lines did not seem to trigger any CI testing failures, I think test coverage is missing for these lines.